test/ExpressiveInstallerTest/RoutersTest.php

```diff
@@ -93,9 +106,27 @@ public function testRouter(
         );
 
         // Test home page
-        $setupRoutes = (strpos($copyFilesKey, 'minimal') !== 0);
+        $setupRoutes = strpos($copyFilesKey, 'minimal') !== 0;
         $response = $this->getAppResponse('/', $setupRoutes);
         $this->assertEquals($expectedResponseStatusCode, $response->getStatusCode());
+
+        /** @var Application $app */
+        $app = $container->get(Application::class);
+        $this->assertCount(count($expectedRoutes), $app->getRoutes());
+        foreach ($app->getRoutes() as $route) {
+            foreach ($expectedRoutes as $expectedRoute) {
+                if ($expectedRoute['name'] === $route->getName()) {
+                    $this->assertEquals($expectedRoute['path'], $route->getPath());
+                    // todo: not sure how we can test middleware?
+                    // $this->assertEquals($expectedRoute['middleware'], $route->getMiddleware());
```

Comment:
Not sure also if splitting `getAppResponse` method into two methods is good, because then we kinda lost isolation for requests. In general, it is possible that the previous request can make some changes and then the next one can be successful or not (because of changes made by previous call - not sure if it is clear, but in real world we are going to have one request per one application build). We assume, the state of the `$app` is not changed after the request, and it's the same as before. I would then keep this method as it is, still we can test all routes and responses, but still it's not gonna tell us, if correct (route) middlewares has been dispatched.